In [2]:
import pandas as pd

In [3]:
movies = pd.read_csv('http://bit.ly/imdbratings') #Above data source is scraped from IMDB website.

In [4]:
movies.head() 

star_rating                     title content_rating   genre  duration  \
0          9.3  The Shawshank Redemption              R   Crime       142   
1          9.2             The Godfather              R   Crime       175   
2          9.1    The Godfather: Part II              R   Crime       200   
3          9.0           The Dark Knight          PG-13  Action       152   
4          8.9              Pulp Fiction              R   Crime       154   

                                         actors_list  
0  [u'Tim Robbins', u'Morgan Freeman', u'Bob Gunt...  
1    [u'Marlon Brando', u'Al Pacino', u'James Caan']  
2  [u'Al Pacino', u'Robert De Niro', u'Robert Duv...  
3  [u'Christian Bale', u'Heath Ledger', u'Aaron E...  
4  [u'John Travolta', u'Uma Thurman', u'Samuel L....

As explained earlier by default it shows top 5 rows. In case Parenthesis any numeric value is used inside parenthesis for example 8 it will show top 8 values. Below is another example with numeric inside parenthesis.

In [6]:
movies.head(8) #Total 8 rows are available and it Indexing starts at 0

star_rating                                          title content_rating  \
0          9.3                       The Shawshank Redemption              R   
1          9.2                                  The Godfather              R   
2          9.1                         The Godfather: Part II              R   
3          9.0                                The Dark Knight          PG-13   
4          8.9                                   Pulp Fiction              R   
5          8.9                                   12 Angry Men      NOT RATED   
6          8.9                 The Good, the Bad and the Ugly      NOT RATED   
7          8.9  The Lord of the Rings: The Return of the King          PG-13   

       genre  duration                                        actors_list  
0      Crime       142  [u'Tim Robbins', u'Morgan Freeman', u'Bob Gunt...  
1      Crime       175    [u'Marlon Brando', u'Al Pacino', u'James Caan']  
2      Crime       200  [u'Al Pacino', u'Robert De Niro', u'Robert Duv...  
3     Action       152  [u'Christian Bale', u'Heath Ledger', u'Aaron E...  
4      Crime       154  [u'John Travolta', u'Uma Thurman', u'Samuel L....  
5      Drama        96  [u'Henry Fonda', u'Lee J. Cobb', u'Martin Bals...  
6    Western       161  [u'Clint Eastwood', u'Eli Wallach', u'Lee Van ...  
7  Adventure       201  [u'Elijah Wood', u'Viggo Mortensen', u'Ian McK...

In [5]:
movies.describe()

star_rating    duration
count   979.000000  979.000000
mean      7.889785  120.979571
std       0.336069   26.218010
min       7.400000   64.000000
25%       7.600000  102.000000
50%       7.800000  117.000000
75%       8.100000  134.000000
max       9.300000  242.000000

describe() aslo ends with parenthesis and as long as DataFrame has one numeric column,it will show descriptive  statistics 

In [10]:
movies.shape

(979, 6)

movies.shape tells that in total there is 979 rows and 6 coloumns 

In [11]:
movies.dtypes

star_rating       float64
title              object
content_rating     object
genre              object
duration            int64
actors_list        object
dtype: object

movies.dtypes provides column properties as we see we have object/string and float and Int data types in our data frame.

In [12]:
type(movies)

pandas.core.frame.DataFrame

Movies is a DataFrame and as DataFrame it has some Methods and Attributes.
###### 1. Methods are the one with parenthesis .
###### 2. Attributes are the one without parenthesis .
Method - Methods are action oriented. Methods basically signifies that some task is to be performed on it
Attributes - Attributes are properties. Attributes are more of properties that tells about DataFrame

Methods comes with parenthesis , Though in most of the cases its not depended on value inside prenthesis but in certain scenario it makes sense.Below is one of example.

In [21]:
movies.describe(include = ['object'])

title content_rating  genre  \
count       979            976    979   
unique      975             12     16   
top     Dracula              R  Drama   
freq          2            460    278   

                                              actors_list  
count                                                 979  
unique                                                969  
top     [u'Daniel Radcliffe', u'Emma Watson', u'Rupert...  
freq                                                    6

###### Part form Bonus Tip

Hitting |shift+Tab | inisde parenthesis provides detail you can hit upto 4 times for multiple results. 